### ISSUE TITLE
Text is not displayed on Android physical device after 0.73.0

### ISSUE THREAD

**Issue by crawraps** [facebook/react-native#42089]
### Description

Since version 0.73.0 text in the Text component is not displayed on a physical connected Android device.

It works well both on Android Studio Emulator (tested with API 33 and API 34) and on physical connected device with react native below 0.73.0

Tested on two different physical devices

If I add background color to it, Text component would be visible, but content is not. Also, text is visible in TextInput component. Text in error logs is not visible too, but visible in loading log (at the very top of the application)

### Steps to reproduce

1. Install the application on physical Android device
2. Notice that any text in the Text component is not displayed

### React Native Version

0.73.1

### Affected Platforms

Runtime - Android

### Output of `npx react-native info`

```text
System:
  OS: Linux 6.6 Arch Linux
  CPU: (12) x64 AMD Ryzen 5 5600H with Radeon Graphics
  Memory: 3.57 GB / 14.98 GB
  Shell:
    version: "5.9"
    path: /usr/bin/zsh
Binaries:
  Node:
    version: 18.18.2
    path: ~/.local/nvm/versions/node/v18.18.2/bin/node
  Yarn: Not Found
  npm:
    version: 9.8.1
    path: ~/.local/nvm/versions/node/v18.18.2/bin/npm
  Watchman:
    version: 20231028.222255.0
    path: /usr/local/bin/watchman
SDKs:
  Android SDK: Not Found
IDEs:
  Android Studio: Not Found
Languages:
  Java:
    version: 17.0.9
    path: /home/crawraps/.local/sdkman/candidates/java/current/bin/javac
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.1
    wanted: 0.73.1
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: false
```
```


### Stacktrace or Logs

```text
-
```

### Reproducer

https://github.com/crawraps/reproducer

### Screenshots and Videos

![screenshot](https://github.com/facebook/react-native/assets/64529130/b1748373-1db0-4f92-a1f7-c6d3129278e4)
initial react native reproducer template

**Comment by cortinico** [comment]
> Install the application on physical Android device

Which device are you using?

**Comment by crawraps** [comment]
> > Install the application on physical Android device
> 
> Which device are you using?

OnePlus 9RT with custom rom riceDroid-13.0
Also tested on Pixel 7 and get same behaviour

**Comment by cortinico** [comment]
That's a screenshot from a Pixel 3 (physical device):

![screen-1703772558](https://github.com/facebook/react-native/assets/3001957/532e6c2a-01e7-471f-ab0b-29bfd437ff9a)

I cannot reproduce what you're reporting @crawraps

**Comment by cetfu** [comment]
![image](https://github.com/facebook/react-native/assets/69468559/11dbe58e-e5fa-4df2-83b0-84e4298f90bf)
Tested on Pixel 5a (Android 14) physical device

**Comment by cortinico** [comment]
Closing as cannot reproduce @crawraps 
Please provide further evidence and we can re-open

**Comment by dhruvpvx** [comment]
@cortinico I'm also facing the same issue in my physical android device it is OnePlus 9r with NameLess Custom Rom.
I think to reproduce this issue someone will need a oneplus device with custom rom

**Comment by dhruvpvx** [comment]
@cortinico currently I'm trying to switch from commits from each release to figure out in which release this issue first appeared
Is there any other way to figure out the issue?

**Comment by dhruvpvx** [comment]
@cortinico I tried switching from commit to commit on 0.73 I was able to generate a build from yarn android but in 72.8 this command is not running. 
it says
 No Metro config found in /Users/Desktop/react-native.

**Comment by cortinico** [comment]
> I'm trying to switch from commits from each release to figure out in which release this issue first appeared

You should use the nightlies version we have listed here:
https://www.npmjs.com/package/react-native?activeTab=versions
to bisect this.

Specifically you can create a new app from a nightly version with:

```
npx react-native@latest init RNNightly --version 0.74.0-nightly-20231031-572dd76ba --skip-install
```

then do `cd RNNightly && yarn && yarn android` and verify when the issue started happening.

**Comment by dhruvpvx** [comment]
as I am able to figure out this is working in 0.72.9 but not it 73.0.

**Comment by dhruvpvx** [comment]
@cortinico this issue first appeared in 0.73.0-nightly-20230521-6d24ee13a working fine in 0.73.0-nightly-20230519-de1e9f5d1

**Comment by dhruvpvx** [comment]
this the commit which is mentioned in the nightly version
https://github.com/facebook/react-native/commit/6d24ee13a47a58ff1b19c5a51f39f52c9a4a8a28
I tried reverting these changes but still text is not visible.

**Comment by cortinico** [comment]
> I tried reverting these changes but still text is not visible.

How have you reverted those changes? The code is pre-compiled on Android, so editing the files inside `node_modules` will have no effect

**Comment by dhruvpvx** [comment]
> > I tried reverting these changes but still text is not visible.
> 
> How have you reverted those changes? The code is pre-compiled on Android, so editing the files inside `node_modules` will have no effect

How can I revert these changes other then that?

**Comment by dhruvpvx** [comment]
hey @cortinico can you please help to check if this is causing the issue

**Comment by cortinico** [comment]
You'll have to use a build from source:
https://reactnative.dev/contributing/how-to-build-from-source

**Comment by dhruvpvx** [comment]
@cortinico this is now working in latest commit I think https://github.com/facebook/react-native/pull/42258 this fixed this issue since all the changes are in the same file as in the commit which first started this issue

**Comment by dhruvpvx** [comment]
@cortinico i'm using react native 0.73 I need this part only how can I get this changing in node_modules doesn't work any other way?

**Comment by cortinico** [comment]
> @cortinico i'm using react native 0.73 I need this part only how can I get this changing in node_modules doesn't work any other way?

Can you try to follow the insutrctions I linked ([reactnative.dev/contributing/how-to-build-from-source](https://reactnative.dev/contributing/how-to-build-from-source)), apply the changes in https://github.com/facebook/react-native/pull/42258 inside your `node_modules` folder, and verify if it solves?

**Comment by Ainias** [comment]
Is the fix included inside RN 0.73.6 or only in the 0.74-RCs?

**Comment by defnotsuhas** [comment]
Hi @cortinico,

I too am facing the same issue. I'm currently upgrading my Project's React Native from 72.7 -> 73.8. The text is being rendered on Android Simulator but on My Physical Device [OnePlus 6T running SuperiorOS Custom Rom Android 13] text is not rendered. I just tried a new project using 0.74 and it's working. The issue is resolved somewhere between 73.8 to 74.0.
